components/org.wso2.carbon.identity.governance/src/test/java/org/wso2/carbon/identity/governance/listener/IdentityMgtEventListenerTest.java

```diff
@@ -0,0 +1,577 @@
+
+    IdentityMgtServiceDataHolder instanceIdentityDataHolder;
+    IdentityMgtServiceDataHolder instance;
+
+    @InjectMocks
+    @Spy
+    IdentityMgtEventListener identityMgtEventListener;
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    @BeforeTest
+    public void setUp() throws Exception {
+        instance = IdentityMgtServiceDataHolder.getInstance();
+        identityEventService = mock(IdentityEventService.class);
+        instance.setIdentityEventService(identityEventService);
+        realmService = mock(RealmService.class);
+        instance.setRealmService(realmService);
+        identityMgtEventListener = new IdentityMgtEventListener();
+        instanceIdentityDataHolder = new IdentityMgtServiceDataHolder();
+    }
+
+    @Test
+    public void testGetExecutionOrderId() {
+        int orderId = identityMgtEventListener.getExecutionOrderId();
+        assertEquals(orderId, 95, "OrderId is not equal to IdentityCoreConstants.EVENT_LISTENER_ORDER_ID");
+
+        IdentityMgtEventListener testIdentityMgtEventListener = spy(IdentityMgtEventListener.class);
+        doReturn(55).when(testIdentityMgtEventListener).getOrderId();
+        orderId = testIdentityMgtEventListener.getExecutionOrderId();
+        assertEquals(orderId, 55, "OrderId is not equal to IdentityCoreConstants.EVENT_LISTENER_ORDER_ID");
+    }
+
+    @DataProvider(name = "dataHandler1")
+    public Object[][] getData1() {
+
+        return new String[][]  {
+                {"admin", "admin"},
+                {"admin", null},
+                {null, "admin"},
+                {null, null}
+        };
+    }
+
+    @Test(expectedExceptions = UserStoreException.class, dataProvider = "dataHandler1")
+    public void testDoPreAuthenticate(String username,
+                                      String pwd) throws Exception {
+        Object credential;
+        if (pwd == null) {
+            credential = null;
+        } else {
+            credential = new String(pwd);
+        }
+        initMocks();
+        assertTrue(identityMgtEventListener.doPreAuthenticate(username, credential, userStoreManager), 
+                "Do pre authenticate flow is failed.");
+
+        doThrow(IdentityEventException.class).when(identityEventService).handleEvent(any(Event.class));
+        identityMgtEventListener.doPreAuthenticate(username, credential, userStoreManager);
+    }
+
+    @DataProvider(name = "dataHandler2")
+    public Object[][] getData2() {
+
+        return new Object[][]  {
+                {"admin", true},
+                {"admin", false},
+                {null, true},
+                {null, false}
+        };
+    }
+
+    @Test(dataProvider = "dataHandler2")
+    public void testDoPostAuthenticate(String userName,
+                                       boolean authenticated) throws Exception {
+        initMocks();
```

Comment:
Can't we `initMocks()` in @BeforeTest method rather than doing it in every test method?